backend/cfg/cfg.ml

```diff
@@ -349,8 +349,57 @@ let is_pure_terminator desc =
     (* CR gyorsh: fix for memory operands *)
     true
 
+let is_pure_operation : operation -> bool = function
+  | Move -> true
+  | Spill -> true
+  | Reload -> true
+  | Const_int _ -> true
+  | Const_float _ -> true
+  | Const_symbol _ -> true
+  | Stackoffset _ -> false
+  | Load _ -> true
+  | Store _ -> false
+  | Intop _ -> true
+  | Intop_imm _ -> true
+  | Negf -> true
+  | Absf -> true
+  | Addf -> true
+  | Subf -> true
+  | Mulf -> true
+  | Divf -> true
+  | Compf _ -> true
+  | Floatofint -> true
+  | Intoffloat -> true
+  | Probe _ -> false
+  | Probe_is_enabled _ -> true
+  | Opaque -> false
+  | Begin_region -> false
+  | End_region -> false
+  | Specific s -> Arch.is_pure_specific s
+  | Name_for_debugger _ -> true
+
```

Comment:
It is pure for https://github.com/ocaml-flambda/flambda-backend/blob/main/backend/amd64/proc.ml#L424,
and my understanding is that its storage is disjoint so that
spilling cannot be observed.

Reply:
Spill (and Reload) are just Move instructions with stack-allocated registers as destination (or source). In particular, allocation of the space on the stack is done during the function prologue, not at the Spill operation.